Support watchdog for frdm_mcxe31b

yvesll opened the pull request:
Use the existing `nxp_s32_swt` driver.

ZhaoxiangJin commented on drivers/watchdog/wdt_mcux_swt.c:
```diff
@@ -0,0 +1,185 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "PERI_SWT.h"
```
Seems this header is not needed?

yvesll committed:
drivers: watchdog: Make SWT compatible with hal_nxp

Avoid multiple declarations of SWT macros.
No need to log error for redundant watchdog disable.

Signed-off-by: Yves Wang <zhengjia.wang@nxp.com>

yvesll committed:
drivers: clock_control: MC_CGM support get SIRC clock rate

Return fixed SIRC clock rate in nxp mc_cgm.

Signed-off-by: Yves Wang <zhengjia.wang@nxp.com>

yvesll committed:
boards: nxp: frdm_mcxce31b allow clear des fes reset flag

On destructive or functional reset, the soc_reset_hook will check reset
source and clear SRAM TCM.
Add support to let developer clear specify reset source to retein memory
content.

Signed-off-by: Yves Wang <zhengjia.wang@nxp.com>

yvesll committed:
dts: nxp: Add watchdog support for MCXE31x

Enable swt as watchdog instance for NXP MCXE31x.

Signed-off-by: Yves Wang <zhengjia.wang@nxp.com>

yvesll committed:
tests: watchdog: Clear watchdog reset source for frdm_mcxe31b

Clear watchdog reset flag then device can retein memory.

Signed-off-by: Yves Wang <zhengjia.wang@nxp.com>

dleach02 commented:
Zephyr Project recently added to CI a checker on dts files. Unfortunately it will validate the whole file and not just the change set to the file.

Please add a separate commit to your PR to do a compliance cleanup of the file so this PR can clear CI.

yvesll commented:
The dts-linter issue was fixed in #98114

dleach02 commented:
> The dts-linter issue was fixed in #98114

Then you will need to rebase when that merges.

ZhaoxiangJin reviewed (approved):
request @manuargue for review since reused s32 driver.
